cloud/libvirt/actuators/machine/utils/domain.go

```diff
@@ -757,3 +766,104 @@ func DomainExists(machine *clusterv1.Machine) (bool, error) {
 
 	return true, nil
 }
+
+// LookupDomainByUUID retrieves a libvirt domain by UUID using the
+// provider config on the given machine.
+func LookupDomainByUUID(machine *clusterv1.Machine, uuid *uuid.UUID) (*libvirt.Domain, error) {
+	machineProviderConfig, err := MachineProviderConfigFromClusterAPIMachineSpec(&machine.Spec)
+	client, err := buildClient(machineProviderConfig.URI)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to build libvirt client: %s", err)
+	}
+
+	virConn := client.libvirt
+	if virConn == nil {
+		return nil, fmt.Errorf(LibVirtConIsNil)
+	}
+
+	dom, err := virConn.LookupDomainByUUID(uuid[:])
+	if err != nil {
+		return nil, fmt.Errorf("Failed to lookup domain by UUID: %v", err)
+	}
+
+	return dom, nil
+}
+
+// UpdateProviderStatus updates the provider status in-place with info
+// from the given libvirt domain.
+func UpdateProviderStatus(status *providerconfigv1.LibvirtMachineProviderStatus, dom *libvirt.Domain) error {
```

Comment:
any machine api specific logic does not belong here, we can address it after merging and rebasing https://github.com/openshift/cluster-api-provider-libvirt/pull/38

Reply:
Fixed.